File: src/server/src/main/java/com/project/application/models/workout/WorkoutDetailsResource.java
package com.project.application.models.workout;

import com.project.domain.workout.Workout;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;
import java.util.UUID;

@Serdeable
public record WorkoutDetailsResource(UUID id, String workoutTemplateName,
                                     List<WorkoutExerciseDetailsResource> exercises) {
    public static WorkoutDetailsResource from(Workout workout) {
        return new WorkoutDetailsResource(workout.getId(),
                workout.getWorkoutTemplate() != null ? workout.getWorkoutTemplate().getName() : null,
                workout.getExercises().isEmpty() ? List.of() : workout.getExercises().stream().map(WorkoutExerciseDetailsResource::from)
                        .toList());
    }
}

File: src/server/src/main/kotlin/com/project/application/models/workout/WorkoutDetailsResource.kt
package com.project.application.models.workout

import com.project.domain.workout.Workout
import io.micronaut.serde.annotation.Serdeable
import java.util.UUID

@Serdeable
data class WorkoutDetailsResource(val id: UUID, val workoutTemplateName: String?) {

    companion object {
        fun from(workout: Workout): WorkoutDetailsResource {
            return WorkoutDetailsResource(workout.id,
                    if (workout.workoutTemplate != null) workout.workoutTemplate!!.name else null)
        }
    }
}
